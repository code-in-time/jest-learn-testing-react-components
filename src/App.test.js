import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

import renderer from 'react-test-renderer';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

// test the snapshot
it('Matched snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot();
}) 
