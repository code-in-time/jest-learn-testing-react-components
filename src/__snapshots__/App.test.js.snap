// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matched snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <div>
    testbcvbcvbc 
    <button
      onClick={[Function]}
      type="button"
    >
      click
    </button>
    <hr />
    <span
      className="text"
    >
      this is text to show
    </span>
  </div>
</div>
`;
